hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/publishing/metric/ThroughputRegistryTest.groovy

```diff
@@ -0,0 +1,65 @@
+package pl.allegro.tech.hermes.frontend.publishing.metric
+
+import com.codahale.metrics.MetricRegistry
+import com.jayway.awaitility.Duration
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry
+import org.assertj.core.api.Assertions
+import pl.allegro.tech.hermes.api.Topic
+import pl.allegro.tech.hermes.common.metric.MetricsFacade
+import pl.allegro.tech.hermes.frontend.metric.ThroughputMeter
+import pl.allegro.tech.hermes.frontend.metric.ThroughputRegistry
+import spock.lang.Shared
+import spock.lang.Specification
+
+import static com.jayway.awaitility.Awaitility.await
+import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.topic
+
+class ThroughputRegistryTest extends Specification {
+    @Shared
+    Topic topicA = topic("group.topicA").build()
+    @Shared
+    Topic topicB = topic("group.topicB").build()
+    @Shared
+    Topic topicC = topic("group.topicC").build();
+
+    private final MetricsFacade metricsFacade = new MetricsFacade(new SimpleMeterRegistry());
+    private final ThroughputRegistry throughputRegistry = new ThroughputRegistry(metricsFacade, new MetricRegistry())
+
+    def "topic throughput should be preserved for instances of the same topic"() {
+        given: "throughput meter for a topic with recorded value"
+        ThroughputMeter meter = throughputRegistry.forTopic(topicA.getName())
+        assert meter.oneMinuteRate == 0.0d
+
+        meter.increment(1024)
+
+        when: "new throughput meter is obtained for the same topic"
+        meter = throughputRegistry.forTopic(topicA.getName())
+
+        then: "throughput is preserved"
+        await().atMost(Duration.TEN_SECONDS).until({
+            Assertions.assertThat(meter.oneMinuteRate).isGreaterThan(0.0d)
```

Comment:
Why do we only check if it is greater than 0 instead of the exact value? I assume we are probably forced to do this because the meter returns inaccurate values

Reply:
yes, it is hard to select the exact value